api/controllers/DashboardController.js

```
@@ -16,275 +14,360 @@ var dashboardController = {
-
-                            var applicationReferences = [];
-
-                            for ( var i=0; i < results.length; i++ ) {
-                                applicationReferences.push(results[i].unique_app_id);
-                            }
+                    type: sequelize.QueryTypes.SELECT,
+                };
+                const userData = HelperService.getUserData(req, res);
+                const displayAppsArgs = {
+                    userData,
+                    totalApplications,
+                    offset,
+                    sortOrder,
+                    currentPage,
+                    searchCriteria,
+                    pageSize,
+                    req,
+                    res,
+                };
+
+                dashboardController._getApplications(
+                    storedProcedureArgs,
+                    displayAppsArgs,
+                    userData.user.electronicEnabled
+                );
+            });
+        });
+    },
 
-                            // Create Request Structure
+    _getApplications(storedProcedureArgs, displayAppsArgs, electronicEnabled) {
+        const applicationType = electronicEnabled
+            ? 'electronic and paper'
+            : 'paper';
+        const queryApplications = dashboardController._chooseStoredProcedure(
+            storedProcedureArgs.secondarySortOrder,
+            electronicEnabled
+        );
+
+        sails.log.info(`Fetching ${applicationType} applications`);
+        sequelize
+            .query(queryApplications, storedProcedureArgs)
+            .then((electronitApplications) => {
+                dashboardController._displayApplications(
+                    electronitApplications,
+                    displayAppsArgs
+                );
+            })
+            .catch((error) => {
+                sails.log(error);
```

this is a silent error - so whatever is calling it wont know if anything failed - is that intended? (or out of scope to deal with right now?_

---

I'm not sure I follow, how did you figure out it's a silent error? Is it because it's missing a log method?
`sails.log.error(error);` instead of `sails.log(error);`

---

silent wasn't the best term sorry 😄, I meant if just logging the error is OK - and does not affect the rest of the flow of the code/there's no need to "recover" from the code you can ignore this. (for example redirecting the user to an error page etc)

---

gotcha, yeah that makes a lot of sense, I've made the updates now.